diff --git a/lib/cjs/index.cjs b/lib/cjs/index.cjs
index 3890ff00a2affac86818f839ef8a23e1b04b937e..c32a965ea3ca4367ba936e3143fdde74990792f5 100644
--- a/lib/cjs/index.cjs
+++ b/lib/cjs/index.cjs
@@ -4197,6 +4197,7 @@ var EvmTransaction = /** @class */ (function () {
                 return EvmInternalTransaction.create(transaction);
             }),
             signature: commonCore.maybe(data.signature, EvmSignature.create),
+            transactionFee: commonCore.maybe(data.transactionFee),
         };
     };
     /**
@@ -4573,6 +4574,13 @@ var EvmTransaction = /** @class */ (function () {
         enumerable: false,
         configurable: true
     });
+    Object.defineProperty(EvmTransaction.prototype, "transactionFee", {
+        get: function () {
+            return this._data.transactionFee;
+        },
+        enumerable: false,
+        configurable: true
+    });
     return EvmTransaction;
 }());
 
@@ -9988,7 +9996,7 @@ var GetWalletHistoryOperation = {
     groupName: "wallets",
     httpMethod: "get",
     routePattern: "/wallets/{address}/history",
-    parameterNames: ["chain", "from_block", "to_block", "from_date", "to_date", "address", "include_internal_transactions", "nft_metadata", "cursor", "order", "limit"],
+    parameterNames: ["chain", "from_block", "to_block", "from_date", "to_date", "address", "include_internal_transactions", "include_input_data", "nft_metadata", "cursor", "order", "limit"],
     hasResponse: true,
     hasBody: false,
     parseResponse: function (json) {
@@ -10002,6 +10010,7 @@ var GetWalletHistoryOperation = {
         var toDate = request.toDate;
         var address = EvmAddress.create(request.address);
         var includeInternalTransactions = request.includeInternalTransactions;
+        var includeInputData = request.includeInputData;
         var nftMetadata = request.nftMetadata;
         var cursor = request.cursor;
         var order = request.order ? EvmOrderList.create(request.order) : undefined;
@@ -10014,6 +10023,7 @@ var GetWalletHistoryOperation = {
             to_date: toDate !== undefined ? toDate.toISOString() : undefined,
             address: address.toJSON(),
             include_internal_transactions: includeInternalTransactions,
+            include_input_data: includeInputData,
             nft_metadata: nftMetadata,
             cursor: cursor,
             order: order ? order : undefined,
@@ -12820,6 +12830,7 @@ function deserializeResponse$7(jsonResponse, request, core) {
         }),
         receiptRoot: jsonResponse.receipt_root,
         receiptStatus: jsonResponse.receipt_status,
+        transactionFee: jsonResponse.transaction_fee,
     });
 }
 
diff --git a/lib/index.d.ts b/lib/index.d.ts
index 3d3af1aaefedd12ea3ffaad9d717a61cf67a447d..28d7fd7d7e4b3abf2293989b21a50bf5a4d59cb9 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -2359,6 +2359,7 @@ interface EvmTransactionData {
     logs: EvmTransactionLog[];
     internalTransactions: EvmInternalTransaction[];
     signature?: EvmSignature;
+    transactionFee?: string;
 }
 
 /**
@@ -2438,6 +2439,7 @@ declare class EvmTransaction {
         blockHash: string;
         receiptRoot?: string | undefined;
         receiptStatus?: number | undefined;
+        transactionFee?: string;
     };
     /**
      * @deprecated This method will be removed soon. To format the value, use one of the properties.
@@ -2655,6 +2657,7 @@ declare class EvmTransaction {
     get v(): number | undefined;
     get r(): string | undefined;
     get s(): string | undefined;
+    get transactionFee(): string | undefined;
 }
 
 /**
@@ -13422,6 +13425,10 @@ interface GetWalletHistoryOperationRequest {
      * @description If the result should contain the internal transactions.
      */
     readonly includeInternalTransactions?: boolean;
+    /**
+     * @description If the result should contain the input data.
+     */
+    readonly includeInputData?: boolean;
     /**
      * @description If the result should contain the nft metadata.
      */
@@ -13447,6 +13454,7 @@ interface GetWalletHistoryOperationRequestJSON {
     readonly to_date?: string;
     readonly address: EvmAddressJSON;
     readonly include_internal_transactions?: boolean;
+    readonly include_input_data?: boolean;
     readonly nft_metadata?: boolean;
     readonly cursor?: string;
     readonly order?: EvmOrderListJSON;
